code/modules/reagents/reagent_dispenser.dm

```diff
@@ -329,3 +337,29 @@
 		to_chat(user, "<span class='notice'>Sprayer refilled.</span>")
 		playsound(get_turf(src), 'sound/effects/refill.ogg', 50, 1, -6)
 		return 1
+		
+/obj/structure/reagent_dispensers/proc/wrenchAnchor(var/mob/user) //proc to wrench a wrenchable dispenser into place
+	for(var/obj/structure/reagent_dispensers/other in loc) //ensure multiple reagent dispensers aren't anchored in one place
+		if(other.anchored == 1 && other.density == 1 && density && !anchored)
+			to_chat(user, "\The [other] is already anchored in this location.")
+			return -1 
+			
+	if(!anchored)
+		if(!istype(src.loc, /turf/simulated/floor)) //Prevent from anchoring shit to shuttles / space
+			if(istype(src.loc, /turf/simulated/shuttle) || istype(src.loc, /turf/space)) //If on the shuttle or empty space
+				to_chat(user, "<span class='notice'>You can't secure \the [src] to [istype(src.loc,/turf/space) ? "space" : "this"]!</span>")
+				return
+
+	user.visible_message(	"[user] begins to [anchored ? "unbolt" : "bolt"] \the [src] [anchored ? "from" : "to" ] the floor.",
+							"You begin to [anchored ? "unbolt" : "bolt"] \the [src] [anchored ? "from" : "to" ] the floor.")
+	playsound(loc, 'sound/items/Ratchet.ogg', 50, 1)
+	if(do_after(user, src, 30))
+		anchored = !anchored
+		if(requiresAnchor) 
+			active = !active
```

Comment:
Is `active` used for anything other than anchoring? Inverting the value will cause the machine to become active when unwrenched if it is already inactive for some other reason.